src/Compilers/CSharp/Portable/Symbols/Source/LocalFunctionSymbol.cs

```diff
@@ -234,13 +236,14 @@ internal void ComputeReturnType()
             var location = _syntax.ReturnType.Location;
             if (_refKind == RefKind.RefReadOnly)
             {
-                DeclaringCompilation.EnsureIsReadOnlyAttributeExists(diagnostics, location, modifyCompilation: false);
+                compilation.EnsureIsReadOnlyAttributeExists(diagnostics, location, modifyCompilation: false);
             }
 
-            if (returnType.NeedsNullableAttribute())
+            if (compilation?.ShouldEmitNullableAttributes(this) == true &&
```

Comment:
>compilation? [](start = 16, length = 12)

Can ```compilation``` ever be null and why other places do not guard?

Reply:
Yes, `compilation` may be null from the EE.

---
In reply to: [293952092](https://github.com/dotnet/roslyn/pull/36398#discussion_r293952092) [](ancestors = 293952092)